driver/web/driver.go

```diff
@@ -97,3 +103,51 @@ func (d *Driver) New(config transaction.Config) (tx transaction.Transaction, err
 	txn.config = config
 	return txn, nil
 }
+
+// monitor the running chrome process to service transactions and recycle at every interval
+func (d *Driver) monitor() (err error) {
+
+	// start chrome process and set process group id to avoid
+	// leaving zombies upon termination
+	d.cmd = exec.Command(d.config.BinaryPath, d.config.BinaryArgs...)
+	d.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	if err := d.cmd.Start(); err != nil {
+		return fmt.Errorf("replicant-executor: error starting chrome process: %w", err)
+	}
+
+	log.Info("chrome process created").Int("pid", int64(d.cmd.Process.Pid)).Log()
+
+	// TODO: don't panic and figure a better way to surface the errors below.
+	go func() {
+		for {
+			select {
+			case <-time.After(d.config.RecycleInterval):
+
+				log.Info("recycling chrome process").Int("pid", int64(d.cmd.Process.Pid)).Log()
+				d.m.Lock()
+
+				// stop chrome process and its children
+				err := syscall.Kill(-d.cmd.Process.Pid, syscall.SIGKILL)
```

Comment:
Shouldn't we try to kill every chrome process instead? There are a lot of error prone areas in this go routine which may leave some processes.

Reply:
Setting the process group id is the correct way to do it. Which error prone areas?

Reply:
(A lot was badly phrased). Take a look at the line below where the go routine just panic and never gets to shut the process down. A lot of these (which might happen for different reasons) will eventually kill the container/machine.